agent/agentssh/agentssh.go

```diff
@@ -348,10 +258,94 @@ func (s *Server) sessionStart(session ssh.Session) (retErr error) {
 	return cmd.Wait()
 }
 
-type readNopCloser struct{ io.Reader }
+// ptySession is the interface to the ssh.Session that startPTYSession uses
+// we use an interface here so that we can fake it in tests.
+type ptySession interface {
```

Comment:
It's a bit annoying that I have to mock out the context methods, but unfortunately, `ssh.Session` is defined to have a method `Context()` that returns `ssh.Context` rather than `context.Context`.

Unfortunately the go type-checker isn't smart enough to understand that `Context() ssh.Context` _must_ also satisfy `Context() context.Context` since `context.Context` is a strict subset of `ssh.Context`.  Alas.